modules/file-system/parallelstore/install_and_mount_parallelstore.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/sh
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -xe # Update this. No need to print command. For debug only.
+
+# Parse access_points, local_mount, mount_options from argument.
+# Format mount-options string to be compatible.
+# e.g. "disable-wb-cache,eq-count=8" --> --disable-wb-cache --eq-count=8.
+for arg in "$@"; do
+	if [[ $arg == --access_points=* ]]; then
+		access_points="${arg#*=}"
+	fi
+	if [[ $arg == --local_mount=* ]]; then
+		local_mount="${arg#*=}"
+	fi
+	if [[ $arg == --mount_options=* ]]; then
+		mount_options="${arg#*=}"
+    mount_options="--${mount_options//,/ --}"
+	fi
+	if [[ $arg == --user_mode=* ]]; then
+		user_mode="${arg#*=}"
+	fi
+done
+
+# To be removed.
+echo "Access_points are..."
+echo "$access_points"
+
+echo "Mount point is..."
+echo "$local_mount"
+
+echo "Mount options is..."
+echo "$mount_options"
+
+echo "User mode is..."
+echo "$user_mode"
+
+## Install the DAOS client library
+## The following commands should be executed on each client vm.
+# For Rocky linux 8,
+if grep -q "ID=\"rocky\"" /etc/os-release && lsb_release -rs | grep -q "8\.[0-9]"; then
+
+	# 1) Add the Parallelstore package repository
+	tee /etc/yum.repos.d/parallelstore-v2-4-el8.repo <<EOF
+[parallelstore-v2-4-el8]
+name=Parallelstore EL8 v2.4
+baseurl=https://us-central1-yum.pkg.dev/projects/parallelstore-packages/v2-4-el8
+enabled=1
+repo_gpgcheck=0
+gpgcheck=0
+EOF
+	dnf makecache
+
+	# 2) Install daos-client
+	dnf install -y epel-release # needed for capstone
+	dnf install -y daos-client
+
+	# 3) Upgrade libfabric
+	dnf upgrade -y libfabric
+
+# For Ubuntu 22.04 and debian 12,
+elif (grep -q "ID=ubuntu" /etc/os-release && lsb_release -rs | grep -q "22\.04") || (grep -q "ID=debian" /etc/os-release && lsb_release -rs | grep -q "12"); then
+
+	# 1) Add the Parallelstore package repository
+	curl https://us-central1-apt.pkg.dev/doc/repo-signing-key.gpg | apt-key add -
+	echo "deb https://us-central1-apt.pkg.dev/projects/parallelstore-packages v2-4-deb main" | tee -a /etc/apt/sources.list.d/artifact-registry.list
+
+	apt update
+
+	# 2) Install daos-client
+	apt install -y daos-client
+
+	# 3) Increase open file limit, for Ubuntu 22.04 only.
+	# This is necessary for dfuse and for the interception library.
+	if grep -q "ID=ubuntu" /etc/os-release && lsb_release -rs | grep -q "22\.04"; then
+		tee -a /etc/security/limits.conf <<EOF
+* soft nofile 131072
+* hard nofile 131072
+EOF
+	fi
+
+	reboot
```

Comment:
I can't wrap my head around this `reboot`. Let's discuss offline.

Reply:
Its a startup script to install drivers for ubuntu and debian. I will ping you offline. For now, this PR focuses on rocky.